extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,273 @@
+            Type _ when targetType == typeof(BlobContainerClient) => blobCollection.Select(b => (BlobClient)b).ToArray(),
+            _ => null
+        };
+
+        private IEnumerable<object>? ToTargetTypeCollection(Type targetType, IEnumerable<object> blobCollection) => targetType switch
+        {
+            Type _ when targetType == typeof(IEnumerable<String>) || targetType == typeof(String[]) => blobCollection.Select(b => (string)b),
+            Type _ when targetType == typeof(IEnumerable<Stream>) => blobCollection.Select(b => (Stream)b),
+            Type _ when targetType == typeof(IEnumerable<Byte[]>) => blobCollection.Select(b => (Byte[])b),
+            Type _ when targetType == typeof(IEnumerable<BlobBaseClient>) => blobCollection.Select(b => (BlobBaseClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobClient>) => blobCollection.Select(b => (BlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlockBlobClient>) => blobCollection.Select(b => (BlockBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<PageBlobClient>) => blobCollection.Select(b => (PageBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<AppendBlobClient>) => blobCollection.Select(b => (AppendBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobContainerClient>) => blobCollection.Select(b => (BlobClient)b),
+            _ => null
+        };
+
+        private async Task<object?> DeserializeToTargetObjectAsync(Type targetType, string connectionName, string containerName, string blobName)
+        {
+            var content = await GetBlobStreamAsync(connectionName, containerName, blobName);
+            return _workerOptions.Value.Serializer.Deserialize(content, targetType, CancellationToken.None);
+        }
+
+        private object? DeserializeToTargetObjectCollection(IEnumerable<object> blobCollection, Type targetType)
+        {
+            (string methodName, Type type) = targetType.IsArray
+                                            ? (nameof(CloneToArray), targetType.GetElementType())
+                                            : (nameof(CloneToList), targetType.GenericTypeArguments[0]);
+
+            blobCollection = blobCollection.Select(b => Convert.ChangeType(b, type));
+            MethodInfo method = typeof(BlobStorageConverter).GetMethod(methodName);
+            MethodInfo genericMethod = method.MakeGenericMethod(type);
+
+            return genericMethod.Invoke(null, new[] { blobCollection.ToList() });
+        }
+
+        private static T[] CloneToArray<T>(IList<object> source)
+        {
+            var result = source.Cast<T>().ToList();
+            return result.ToArray();
+        }
+
+        private static IEnumerable<T> CloneToList<T>(IList<object> source)
+        {
+            return source.Cast<T>().ToList();
```

Comment:
Is the purpose of this to just change IList<object> to IList<T> - does it matter too much? What happens if we return IList<object> to users?

Reply:
That depends how this is used. If this is for a function parameter binding then whatever we return must be assignable to the declared parameter type. If they declare `IEnumerable<SomeType> input` then `IList<object>` is **not** assignable and would throw. The `T` must be `SomeType` or a more derived class of `SomeType`.

Reply:
Yes, seeing exception similar to what Jacob has mentioned.